File: Difficulty: Medium/Bottom View of Binary Tree/bottom-view-of-binary-tree.java
class Solution
{
    //Function to return a list containing the bottom view of the given tree.
    public ArrayList <Integer> bottomView(Node root)
    {
        // Code here
        ArrayList<Integer> ans=new ArrayList<Integer>();
        if(root==null){
            return ans;
        }
        Map<Integer,Integer> map=new TreeMap<>();
        Queue<Node> q=new LinkedList<Node>();
        root.hd=0;
        q.add(root);
        while(!q.isEmpty()){
            Node temp=q.remove();
            int hd=temp.hd;
            map.put(hd,temp.data);
            if(temp.left!=null){
                temp.left.hd=hd-1;
                q.add(temp.left);
            }
            if(temp.right!=null){
                temp.right.hd=hd+1;
                q.add(temp.right);
            }
        }
          for(Integer val:map.values()){
            ans.add(val);
        }
        
        return ans;
        
        
    }
}

File: Difficulty: Medium/Bottom View of Binary Tree/bottom-view-of-binary-tree.py
class Solution:
    def bottomView(self, root):
        # code here
        from collections import deque
        
        if not root:
            return []

        # Dictionary to store the last node at each horizontal distance
        hd_node_map = {}

        # Queue for level order traversal: stores pairs of (node, horizontal distance)
        queue = deque([(root, 0)])

        while queue:
            node, hd = queue.popleft()
            # Overwrite the value at horizontal distance with the current node
            hd_node_map[hd] = node.data

            if node.left:
                queue.append((node.left, hd - 1))
            if node.right:
                queue.append((node.right, hd + 1))

        # Extract the bottom view from the map, sorted by horizontal distance
        bottom_view = [hd_node_map[hd] for hd in sorted(hd_node_map)]
        return bottom_view